src-built-in/components/UIAPI/reducers/workspaces.ts

```diff
@@ -0,0 +1,27 @@
+import { WorkspaceState, WorskpaceActions, ActionTypes } from '../types';
+import withLogging from '../hoReducers/logging';
+
+export const initialState: WorkspaceState = {
+	activeWorkspace: {
+		name: null
+	}
+};
+
+export const workspaces = (state: WorkspaceState = initialState, action: WorskpaceActions) => {
+	const { type, payload } = action;
+	switch (type) {
+		case ActionTypes.SET_ACTIVE_WORKSPACE:
+			return {
+				...state,
+				activeWorkspace: {
+					name: payload.name
+				}
+			}
+		default:
+			return state;
+	}
+}
+
+// Wraps the reducer with some logging so that we can debug
+// field reported bugs.
+export default withLogging("Workspaces", workspaces);
```

Comment:
We don't need to do this here. It only needs to be done once at the top level when we create the store.

Reply:
what? I'm going to merge this because you guys approved it but please explain.